### PR TITLE
Save trait state

### PR DESCRIPTION
**Pull request by yongpingchen**
- Add `firmwareVersion` parameter to onboard method
- Add `save(JsonObject, boolean)` to save thing state for trait use case. 
- Add test to save trait state, including onboard.

### CODE REVIEW

File: src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java [hunk 1]
```diff
@@ -138,6 +138,65 @@ public KiiThing onboard(String thingID, String thingPassword) throws KiiRestExce
...
+			throw new IllegalArgumentException("vendorThingID is null");
+		}
+		if (thingPassword == null) {
+			throw new IllegalArgumentException("thingPassword is null");
+		}
+
+		Map<String, String> headers = this.newAuthorizedHeaders();
+
+		JsonObject requestBody;
+		if (options != null) {
```
Comment (nfukuzaki) [inline src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java]: unnecessary

File: src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java [hunk 2]
```diff
@@ -138,6 +138,65 @@ public KiiThing onboard(String thingID, String thingPassword) throws KiiRestExce
...
+		requestBody.addProperty("owner", Prefix.USER.getValue() + getCredentials().getID());
+
+		KiiRestRequest request = new KiiRestRequest(getUrl("/onboardings"), Method.POST, headers, MEDIA_TYPE_ONBOARDING_WITH_VENDOR_THING_ID_BY_OWNER_REQUEST, requestBody);
+		try {
+			Response response = this.execute(request);
+			JsonObject responseBody = this.parseResponseAsJsonObject(request, response);
+			JsonObject thingJson = new JsonObject();
+			thingJson.addProperty(
+					KiiThing.PROPERTY_THING_ID.getName(),
+					responseBody.get("thingID").getAsString());
```
Comment (nfukuzaki) [inline src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java]: use KiiThing.PROPERTY_THING_ID.get(responseBody)

File: src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java [hunk 3]
```diff
@@ -138,6 +138,65 @@ public KiiThing onboard(String thingID, String thingPassword) throws KiiRestExce
...
+			return new KiiThing(thingJson)
+					.setAccessToken(responseBody.get("accessToken").getAsString());
```
Comment (nfukuzaki) [inline src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java]: use KiiThing.PROPERTY_ACCESS_TOKEN.get(responseBody)

Comment (nfukuzaki) [inline src/main/java/com/kii/cloud/rest/client/resource/KiiThingIfResource.java]: I'm not sure correct key is "accessToken" or "_accessToken".

**Comment by yongpingchen** [comment]
@nfukuzaki fixed. Please confirm

**Pull request merged**